code/src/components/AddCategory.js

```diff
@@ -0,0 +1,24 @@
+import React, { useState } from "react"
+import { useDispatch } from 'react-redux'
+import { todos } from '../reducers/todos'
+import { InputForm, TextInput, PostButton } from '../lib/InputForm'
+
+export const AddCategory = () => {
+    const [newCategory, setNewCategory] = useState()
+    const dispatch = useDispatch()
+    const handleSubmit = (event) => {
+        event.preventDefault()
+        dispatch(
+            todos.actions.addCategory({ itemInfo: { name: newCategory, reveal: false, items: [] } })
+        )
+        setNewCategory('')
```

Comment:
make sure to check if the input is empty before saving, either here in the handleSubmit or if you set a minLength on the input field. Right now you can save a category without a name